cmds/id/id.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+// Print process information.
+//
+// Synopsis:
+//      id [-gGnu]
+//
+// Description:
+//      id displays the uid, guid and groups of the calling process
+// Options:
+//  		-g, --group     print only the effective group ID
+//		  -G, --groups    print all group IDs
+//		  -n, --name      print a name instead of a number, for -ugG
+//		  -u, --user      print only the effective user ID
+//
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+var (
+	GROUP_FILE  = "/etc/group"
+	PASSWD_FILE = "/etc/passwd"
+	l           = log.New(os.Stderr, "", 0)
```

Comment:
I think Anthony has a good point here, I wonder if we should not just do this everywhere to get better looking output. It would make writing tests a lot easier.

As for the file name constants: these names have been constant for 45 years. I would suggest dropping the constant and just using the name inline. They're not ever going to change.